src/main/java/org/mule/extension/oauth2/internal/AbstractGrantType.java

```diff
@@ -303,4 +305,15 @@ public boolean isDefaultTokenManager() {
     return defaultTokenManager || tokenManager == null;
   }
 
+  public boolean readsAuthenticatedResponseBody() {
```

Comment:
precalculate the value this method returns at initialization, since the expression won't change between requests